src/Payload.php

```diff
@@ -60,6 +60,41 @@ public function getClaims()
     }
 
     /**
+     * Checks if a payload matches some expected values.
+     *
+     * @param  array  $values
+     * @param  bool  $strict
+     *
+     * @return bool
+     */
+    public function matches(array $values, $strict = false)
+    {
+        if (empty($values)) {
+            return false;
+        }
+
+        foreach ($values as $key => $value) {
+            if (($strict && $this->get($key) !== $value) || $this->get($key) != $value) {
```

Comment:
doesn't this negate the point of having a strict mode?

maybe something like this would be better? - https://github.com/illuminate/support/blob/5.2/Collection.php#L222-L228